src/cloudant/index.py

```diff
@@ -40,18 +40,21 @@ class Index(object):
         Index.
     :param str design_document_id: Optional identifier of the design document.
     :param str name: Optional name of the index.
+    :param bool partitioned: Optional. Create as a partitioned index. Defaults
+        to ``False``.
```

Comment:
Just to check - this implies that we don't follow the server default of `true` if the database is partitioned and `false` otherwise. I'm OK with that as long as it is clear and it looks like the code matches up, but I just wanted to call out the subtle behaviour difference.

Reply:
Ah, yes. I overlooked this subtlety.

The default is `False`. The database object is passed to the index constructor so I could get the `partitioned` value and use that as a default if preferred. But if you're happy then I'll leave it alone. The current implementation matches java-cloudant too.

Reply:
We probably need to more explicitly note that the behaviour is different to avoid any confusion.
I think most "real production" use cases aren't creating indexes and design docs on the fly, they are version controlled and deployed so as to avoid index re-builds and so on, so I don't think it is critical that this matches the server as long as we are clear about what is happening.

Reply:
Updated docstring for extra clarity:
> Defaults to ``False`` for both partitioned and non-partitioned databases.